### PR TITLE
Vue web update

### PR DESCRIPTION
**Pull request by ErisMik**

### CODE REVIEW

**Review by laves — APPROVED** [review]

**Review by mrrostam — APPROVED** [review]

File: binding/vue/src/rhino.ts [hunk 1]
```diff
@@ -31,17 +27,17 @@ import { RhinoContext, RhinoInference, RhinoWorkerFactory } from '@picovoice/rhi
...
-    inferenceCallback: (inference: RhinoInference) => void,
+    accessKey: string,
+    context: RhinoContext,
+    inferenceCallback: InferenceCallback,
+    model: RhinoModel,
+    options: RhinoOptions,
     contextCallback: (info: string) => void,
-    readyCallback: () => void,
+    isLoadedCallback: (isLoaded: boolean) => void,
+    isListeningCallback: (isListening: boolean) => void,
```
Comment (ksyeo1010) [inline binding/vue/src/rhino.ts:37]: based on what I did on Vue, this types have to be a variable in line 51.

**Review by ksyeo1010 — APPROVED** [review]

**Review by ksyeo1010 — APPROVED** [review]

File: binding/vue/README.md [hunk 2]
```diff
@@ -1,48 +1,38 @@
...
-- Vue.js 3.0.0+
-
-The Picovoice SDKs for Web are powered by WebAssembly (WASM), the Web Audio API, and Web Workers.
-
-All modern browsers (Chrome/Edge/Opera, Firefox, Safari) are supported, including on mobile. Internet Explorer is _not_ supported.
-
-Using the Web Audio API requires a secure context (HTTPS connection) - except `localhost` - for local development.
+- Chrome / Edge
+- Firefox
+- Safari
```
Comment (laves) [inline binding/vue/README.md:35]: can you call this section `Browser Compatibility` and add a section called `Framework Compatibility` where you add:
```
- Vue.js 2.6.11+
- Vue.js 3.0.0+
```

File: binding/vue/README.md [hunk 3]
```diff
@@ -52,154 +42,198 @@ Signup or Login to [Picovoice Console](https://console.picovoice.ai/) to get you
...
+### Initialize Rhino
+
+**Note**: Due to limitations on Vue, one component can only have one instance of Rhino. To use multiple instances of Rhino instead, check out [Rhino Web binding](https://www.npmjs.com/package/@picovoice/rhino-web).
+
+Create the following functions:
+ - `inferenceCallback` function to get the streaming results from the worker
+ - `contextCallback` function to get the context info from Rhino once the owrker is loaded
+ - `isLoadedCallback` function to check if Rhino has loaded
+ - `isListeningCallback` function to check if Rhino is listening for an inference
+ - `errorCallback` function to catch any error occurred
```
Comment (laves) [inline binding/vue/README.md:149]: the wording here makes it sounds like the user has to call them to check:
```suggestion
 - `inferenceCallback` streams inference results from Rhino
 - `contextCallback` function that delivers context info when Rhino has completed initialization
 - `isLoadedCallback` function that executes when Rhino has loaded or unloaded
 - `isListeningCallback` function that executes when Rhino is listening for an inference
 - `errorCallback` function to catch any error occurred
```

File: binding/vue/README.md [hunk 4]
```diff
@@ -52,154 +42,198 @@ Signup or Login to [Picovoice Console](https://console.picovoice.ai/) to get you
...
-      factoryArgs: {
-        accessKey: '${ACCESS_KEY}',  // AccessKey obtained from Picovoice Console (https://console.picovoice.ai/)
-        context: {
-          base64: `RHINO_TRAINED_CONTEXT_BASE_64_STRING`
-        },
-      }
-    };
+...
+methods: {
+  keywordDetectionCallback: function(inference) {
```
Comment (laves) [inline binding/vue/README.md:154]:
```suggestion
  inferenceCallback: function(inference) {
```

File: binding/vue/README.md [hunk 5]
```diff
@@ -52,154 +42,198 @@ Signup or Login to [Picovoice Console](https://console.picovoice.ai/) to get you
...
+the `base64` model.
+
+## Non-English Languages
+
+In order to detect non-English inferences you need to use the corresponding model file (`.pv`). The model files for all
+supported languages are available [here](https://github.com/Picovoice/rhino/tree/master/lib/common).
+
+## Demo
+
+For example usage refer to our [Web demo application](https://github.com/Picovoice/rhino/tree/master/demo/web).
```
Comment (laves) [inline binding/vue/README.md:239]:
```suggestion
For example usage, refer to our [Vue demo application](https://github.com/Picovoice/rhino/tree/master/demo/vue).
```

File: binding/vue/src/rhino.ts [hunk 6]
```diff
@@ -51,117 +50,94 @@ export default {
...
-            return false;
+        errorCallback: function (): void {
+          return;
+        },
+        async init(
+          accessKey: string,
+          context: RhinoContext,
+          inferenceCallback: InferenceCallback,
+          model: RhinoModel,
+          contextCallback: (info: string) => void,
```
Comment (laves) [inline binding/vue/src/rhino.ts:68]: I think this should be called contextInfoCallback. All of our other SDKs call the property contextInfo, so this callback should also

File: binding/vue/src/rhino.ts [hunk 7]
```diff
@@ -51,117 +50,94 @@ export default {
...
+          isLoadedCallback: (isLoaded: boolean) => void ,
+          isListeningCallback: (isListening: boolean) => void ,
+          errorCallback: (error: any) => void,
+          options: RhinoOptions = {}
+        ): Promise<void> {
+          if (options.processErrorCallback) {
+            console.warn("'processErrorCallback' options is not supported, use 'errorCallback' instead.");
```
Comment (laves) [inline binding/vue/src/rhino.ts:75]: can you update this to the message that's used in the other SDKs?

File: demo/vue/README.md [hunk 8]
```diff
@@ -1,14 +1,44 @@
-# rhino-web-vue-demo
+# Rhino demo for Rect
```
Comment (laves) [inline demo/vue/README.md:1]:
```suggestion
# Rhino demo for Vue
```

File: demo/vue/README.md [hunk 9]
```diff
@@ -1,14 +1,44 @@
...
+
+Rhino is:
 
-The widget shows the various loading and error events, as well as mounting/unmounting the `VoiceWidget` with a toggle, demonstrating the complete lifecycle of Rhino with in a Vue app.
+* using deep neural networks trained in real-world environments.
+* compact and computationally-efficient, making it perfect for IoT.
+* self-service. Developers and designers can train custom models using [Picovoice Console](https://console.picovoice.ai/).
 
-This project was bootstrapped with Vue CLI. See the [Configuration Reference](https://cli.vuejs.org/config/).
+## Compatibility
```
Comment (laves) [inline demo/vue/README.md:31]: add note RE: Vue2/Vue3 compatibility (you can copy the one Kwangsoo and I used)

**Review by laves — CHANGES_REQUESTED** [review]

File: binding/vue/README.md [hunk 10]
```diff
@@ -52,154 +47,198 @@ Signup or Login to [Picovoice Console](https://console.picovoice.ai/) to get you
...
+yarn add @picovoice/rhino-vue @picovoice/web-voice-processor
+```
+
+or using `npm`:
 
 ```console
-yarn add @picovoice/rhino-web-vue @picovoice/rhino-web-en-worker @picovoice/web-voice-processor
+npm install --save @picovoice/rhino-vue @picovoice/web-voice-processor
 ```
+### AccessKey
```
Comment (laves) [inline binding/vue/README.md:61]: Duplicate AccessKey section (you may want to double check your other bindings for this)

**Review by laves — CHANGES_REQUESTED** [review]

**Review by laves — APPROVED** [review]

**Pull request merged**